crates/re_space_view_spatial/src/visualizers/entity_iterator.rs

```diff
@@ -237,3 +237,45 @@ macro_rules! impl_process_archetype {
 seq!(NUM_COMP in 0..10 {
     impl_process_archetype!(for N=1, M=NUM_COMP);
 });
+
+/// Count the number of primary instances for a given archetype query that should be displayed.
+///
+/// Returned value might be conservative and some of the instances may not be displayable after all,
+/// e.g. due to invalid transformation etc.
+pub fn count_instances_in_archetype_views<System: IdentifiedViewSystem, A: Archetype>(
+    ctx: &ViewerContext<'_>,
+    query: &ViewQuery<'_>,
+) -> usize {
+    // TODO(andreas/cmc): Use cached code path for this.
+    // This is right now a bit harder to do and requires knowing all queried components.
+    // The only thing we really want to pass here are the POV components.
```

Comment:
cc @teh-cmc maybe this is easier than I thought? Had trouble using the cache here and figured it's not that important anyways for my usecases (profiler says it really isn't)

Reply:
If you're not deserializing nor joining anything you are much better off with a raw query anyway :+1: